plugins/field-colour-hsv-sliders/src/field_colour_hsv_sliders.ts

```diff
@@ -0,0 +1,639 @@
+   * the minimum and maximum control points should be.
+   */
+  static readonly THUMB_RADIUS: number = 12;
+  /* eslint-enable @typescript-eslint/naming-convention */
+
+  /** Helper colour structures to allow manipulation in the HSV colour space. */
+  private static readonly helperHsv: HsvColour = new HsvColour();
+
+  /** Helper colour structures to support conversion to the RGB colour space. */
+  private static readonly helperRgb: RgbColour = new RgbColour();
+
+  /** Array holding info needed to unbind events. Used for disposing. */
+  private boundEvents: Blockly.browserEvents.Data[] = [];
+
+  /** HTML span element to display the current hue. */
+  private hueReadout: HTMLSpanElement | null = null;
+
+  /** HTML range input element for editing hue. */
+  private hueSlider: HTMLInputElement | null = null;
+
+  /** HTML span element to display the current saturation. */
+  private saturationReadout: HTMLSpanElement | null = null;
+
+  /** HTML range input element for editing saturation. */
+  private saturationSlider: HTMLInputElement | null = null;
+
+  /** HTML span element to display the current brightness. */
+  private brightnessReadout: HTMLSpanElement | null = null;
+
+  /** HTML range input element for editing brightness. */
+  private brightnessSlider: HTMLInputElement | null = null;
+
+  /** HTML div element containing all the labels and sliders. */
+  private dropdownContainer: HTMLDivElement | null = null;
+
+  /**
+   * Class for an HSV colour sliders field.
+   * @param value The initial value of the field. Should be in '#rrggbb' format.
+   *     Also accepts Field.SKIP_SETUP if you wish to skip setup (only used by
+   *     subclasses that want to handle configuration and setting the field
+   *     value after their own constructors have run).
+   * @param validator A function that is called to validate changes to the
+   *     field's value. Takes in a colour string & returns a validated colour
+   *     string ('#rrggbb' format), or null to abort the change.
+   * @param config A map of options used to configure the field. See the [field
+   *     creation documentation]{@link
+   *     https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/colour}
+   *     for a list of properties this parameter supports.
+   */
+  constructor(
+      value?: string | Sentinel,
+      validator?: FieldColourValidator,
+      config?: FieldColourConfig) {
+    super(value, validator, config);
+  }
+
+  /* eslint-disable @typescript-eslint/naming-convention */
+  /**
+   * Create and show the colour field's editor.
+   * @override
+   */
+  protected showEditor_(): void {
+    /* eslint-enable @typescript-eslint/naming-convention */
+    this.createDropdownSliders();
+    if (!this.dropdownContainer || !this.hueSlider) {
+      throw new Error('Failed to initialize the HSV sliders.');
+    }
+    Blockly.DropDownDiv.getContentDiv().appendChild(this.dropdownContainer);
```

Comment:
Maybe import DropDownDiv directly vs accessing through the Blockly namespace?

Reply:
Hm, actually I'm not sure how to do that in this context. If I add the import:
import {DropDownDiv} from 'blockly/core/dropdowndiv';
Upong running the test I get the error:
Error: Cannot find module 'blockly/core/dropdowndiv'

Reply:
Oh wait nevermind, I'm not sure if it's possible outside of core.